### PR TITLE
Support role-based access control (only pb)

### PR DESCRIPTION
**Pull request by SimFG**
#16986 
Signed-off-by: SimFG <bang.fu@zilliz.com>

### CODE REVIEW

File: internal/distributed/proxy/client/client.go [hunk 1]
```diff
@@ -209,3 +209,16 @@ func (c *Client) ClearCredUsersCache(ctx context.Context, req *internalpb.ClearC
...
 }
+
+func (c *Client) RefreshPolicyInfoCache(ctx context.Context, req *proxypb.RefreshPolicyInfoCacheRequest) (*commonpb.Status, error) {
+	ret, err := c.grpcClient.ReCall(ctx, func(client interface{}) (interface{}, error) {
+		if !funcutil.CheckCtxValid(ctx) {
+			return nil, ctx.Err()
+		}
+		return client.(proxypb.ProxyClient).RefreshPolicyInfoCache(ctx, req)
+	})
+	if err != nil || ret == nil {
```
Comment (jaime0815) [inline internal/distributed/proxy/client/client.go:220]: err may be nil if ret is nil?

File: internal/distributed/rootcoord/client/client.go [hunk 2]
```diff
@@ -612,3 +612,198 @@ func (c *Client) ListCredUsers(ctx context.Context, req *milvuspb.ListCredUsersR
...
+
+func (c *Client) CreateRole(ctx context.Context, req *internalpb.RoleInfo) (*commonpb.Status, error) {
+	ret, err := c.grpcClient.ReCall(ctx, func(client interface{}) (interface{}, error) {
+		if !funcutil.CheckCtxValid(ctx) {
+			return nil, ctx.Err()
+		}
+		return client.(rootcoordpb.RootCoordClient).CreateRole(ctx, req)
+	})
+	if err != nil || ret == nil {
+		return nil, err
```
Comment (jaime0815) [inline internal/distributed/rootcoord/client/client.go:624]: err maybe nil value?

File: internal/proto/common.proto [hunk 3]
```diff
@@ -42,7 +44,19 @@ enum ErrorCode {
     DeleteCredentialFailure = 31;
     GetCredentialFailure = 32;
     ListCredUsersFailure = 33;
-    NotShardLeader = 34;
+    GetCurrentUserFailure = 34;
```
Comment (jaime0815) [inline internal/proto/common.proto:47]: GetUserFailure

File: internal/proto/common.proto [hunk 4]
```diff
@@ -42,7 +44,19 @@ enum ErrorCode {
...
+    CreateRoleFailure = 35;
+    RoleListFailure = 36;
```
Comment (jaime0815) [inline internal/proto/common.proto:49]: the first word of the variable use a verb within 49-50 and 52-53 lines

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

File: internal/distributed/proxy/service.go [hunk 5]
```diff
@@ -844,3 +844,78 @@ func (s *Server) DeleteCredential(ctx context.Context, req *milvuspb.DeleteCrede
 func (s *Server) ListCredUsers(ctx context.Context, req *milvuspb.ListCredUsersRequest) (*milvuspb.ListCredUsersResponse, error) {
 	return s.proxy.ListCredUsers(ctx, req)
 }
+
+func (s *Server) CreateRole(ctx context.Context, request *milvuspb.CreateRoleRequest) (*commonpb.Status, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (s *Server) RoleList(ctx context.Context, request *milvuspb.RoleListRequest) (*milvuspb.RoleListResponse, error) {
```
Comment (xiaofan-luan) [inline internal/distributed/proxy/service.go:853]: Raname to ListRole, and all the listed operation should follow~

File: internal/distributed/proxy/service.go [hunk 6]
```diff
@@ -844,3 +844,78 @@ func (s *Server) DeleteCredential(ctx context.Context, req *milvuspb.DeleteCrede
...
+	//TODO implement me
+	panic("implement me")
+}
+
+func (s *Server) ResourceList(ctx context.Context, request *milvuspb.ResourceListRequest) (*milvuspb.ResourceListResponse, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (s *Server) PrivilegeListOfResource(ctx context.Context, request *milvuspb.PrivilegeListOfResourceRequest) (*milvuspb.PrivilegeListOfResourceResponse, error) {
```
Comment (xiaofan-luan) [inline internal/distributed/proxy/service.go:893]: DescribeResource

File: internal/distributed/proxy/service.go [hunk 7]
```diff
@@ -844,3 +844,78 @@ func (s *Server) DeleteCredential(ctx context.Context, req *milvuspb.DeleteCrede
...
+	//TODO implement me
+	panic("implement me")
+}
+
+func (s *Server) RevokePrivilege(ctx context.Context, request *milvuspb.PrivilegeRequest) (*commonpb.Status, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func (s *Server) PrincipalGrantList(ctx context.Context, request *milvuspb.GrantListRequest) (*milvuspb.GrantListResponse, error) {
```
Comment (xiaofan-luan) [inline internal/distributed/proxy/service.go:908]: what is PrincipalGrantList?

**Pull request merged**